src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -68,6 +68,16 @@ public override BoundNode VisitConversion(BoundConversion node)
 
             var rewrittenType = VisitType(node.Type);
 
+            // special handling for initializers converted to ROS<T>
+            if (node.Operand is BoundConvertedStackAllocExpression stackAllocExpression &&
+                TypeSymbol.Equals(rewrittenType.OriginalDefinition, _compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.ConsiderEverything) &&
+                _compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_RuntimeHelpers__CreateSpanRuntimeFieldHandle) is not null &&
+                CodeGen.CodeGenerator.UseCreateSpanForReadOnlySpanStackAlloc(stackAllocExpression.ElementType, stackAllocExpression.InitializerOpt, isEncDelta: this.EmitModule?.IsEncDelta == true))
+            {
+                var count = VisitExpression(stackAllocExpression.Count);
```

Comment:
>var count = VisitExpression(stackAllocExpression.Count); [](http://example.com/codeflow?start=16&length=56)

All expressions should be visited and the ```UseCreateSpanForReadOnlySpanStackAlloc``` should probably be performed on the lowered form of the initializer.